voicemail-api/javascript/index.html

```diff
@@ -0,0 +1,231 @@
+                onchange="getQueueVoicemails(this)">
+            </select>
+        </div>
+    </div>
+
+    <div class="row row-no-gutters" style="margin: 25px;">
+        <div id="divListContainer" style="margin: 25px;">
+            <ul class="list-group list-group-horizontal">
+                <li class="list-group-item download-link"><b>Download Link</b></li>
+                <li class="list-group-item other-rows"><b>Duration</b></li>
+                <li class="list-group-item other-rows"><b>ANI</b></li>
+                <li class="list-group-item other-rows"><b>Date and Time</b></li>
+            </ul>
+        </div>
+    </div>
+</body>
+</html>
+
+<script>
+    const clientId = process.env.GENESYS_CLOUD_CLIENT_ID;
+    const redirectUri = window.location.href;
+
+    // Set Genesys Cloud objects
+    const platformClient = require('platformClient');
+    const client = platformClient.ApiClient.instance;
+    const voicemailApi = new platformClient.VoicemailApi();
+    const routingApi = new platformClient.RoutingApi();
+
+    let voicemailDetails = "";
+
+    // Set Genesys Cloud settings
+    client.setEnvironment('mypurecloud.com');
+    client.setPersistSettings(true, 'test_app');
+
+    // Accessing Genesys Cloud objects
+    $(document).ready(() => {
+        client.loginImplicitGrant(clientId, redirectUri)
+            .then(() => {
+                console.log('Logged in');
+            })
+    });
+
+    // Identify user's selected source of voicemail
+    function identifySelect(id) {
+        let selection = id[id.selectedIndex].id;
+        return selection == "queue" ? getQueueList() :
+            getMeVoicemails();
+    }
+    
+    // Display queues available associated in user credentials
+    function getQueueList() {
+        $("#divQueue").show();
+        let opts = {
+            'joined': true, // Boolean | Joined
+            'pageSize': 100, // Number | Page size
+            'pageNumber': 1, // Number | Page number
+            'sortBy': "name", // String | Sort by
+            'sortOrder': "asc" // String | Sort order
+        };
+
+        routingApi.getRoutingQueuesMe(opts)
+            .then((queueData) => {
+                console.log(`getRoutingQueuesMe success! data: ${JSON.stringify(queueData, null, 2)}`);
+                let queueDetails = queueData.entities;
+                queueDetails.forEach(data => createQueueList(data.id, data.name));
+            })
+            .catch((err) => {
+                console.log('There was a failure calling getRoutingQueuesMe');
+                console.error(err);
+            });
+    }
+
+    // Get all voicemail details associated with current user 
+    function getMeVoicemails() {
+        let opts = {
+            'pageSize': 100, // Number | Page size
+            'pageNumber': 1 // Number | Page number
+        };
+
+        voicemailApi.getVoicemailMeMessages(opts)

```

Comment:
@PrinceNebulon Is there a reason why we call things getVoicemailMeMessages on the routing APIs instead of getMyVoiceMailMessages().  Seems a little weird, but I have seen it in a couple of places in different parts of the API.

Reply:
Operation names (SDK method names) come from the swagger definition and are determined by the Public API. [getVoicemailMeMessages](https://developer.mypurecloud.com/api/rest/client-libraries/javascript/VoicemailApi.html#getVoicemailMeMessages) uses "me" because that's what's in the resource path (`GET /api/v2/voicemail/me/messages`). The logic to generate these operations can be changed, but not for the existing SDKs as that's a significant breaking change. Defining new generation logic is something that could be done for future iterations of the SDKs.